tests/unit/api/apiUtils/object/abortMultipartUpload.spec.js

```
@@ -0,0 +1,677 @@
+                        assert.strictEqual(err, undefined); // Should continue despite deletion error
+
+                        // Verify the error was logged
+                        const errorLogCalls = logErrorSpy.getCalls().filter(call =>
+                            call.args[0] === 'error deleting object metadata'
+                        );
+                        assert.strictEqual(errorLogCalls.length, 1,
+                            'Should log error when object metadata deletion fails');
+                        assert.strictEqual(errorLogCalls[0].args[1].error.is.InternalError, true);
+
+                        // Cleanup
+                        logErrorSpy.restore();
+                        metadataValidateMultipartStub.restore();
+                        getMPUpartsStub.restore();
+                        batchDeleteObjectMetadataStub.restore();
+                        deleteObjectMDStub.restore();
+                        done();
+                    }, abortRequest);
+                });
+            });
+        });
+    });
+
+    describe('error handling', () => {
+        it('should handle external data abort error', done => {
+            dataAbortMPUStub.restore();
+            sinon.stub(data, 'abortMPU').callsFake((objectKey, uploadId, location, bucketName,
+                request, destBucket, locationConstraintCheckFn, log, callback) => {
+                callback(errors.InternalError);
+            });
+
+            createBucketAndMPU(false, (err, uploadId) => {
+                assert.ifError(err);
+
+                abortMultipartUpload(authInfo, bucketName, objectKey, uploadId, log, err => {
+                    assert(err);
+                    assert.strictEqual(err.is.InternalError, true);
+                    done();
+                }, abortRequest);
+            });
+        });
+
+        it('should continue despite data deletion errors', done => {
```

I don't know if it make sense here but the log is very important here (as we have a silent error). We want 1. to be sure that we continue even if we have an error 2. but also to have a log to be able to debug and understand. A "good" practice can be to check that we called the log function. We can avoid a refactor later that will mute this error. It's not useful to test all logs but I think here it's a good one ?